### PR TITLE
plugins/nvim-web-devicons: init

### PR DESCRIPTION
**Pull request by refaelsh**
The Nixpkg plugin is here: https://search.nixos.org/packages?channel=24.05&show=vimPlugins.nvim-web-devicons&from=0&size=50&sort=relevance&type=packages&query=++nvim-web-devicons.

https://github.com/nvim-tree/nvim-web-devicons/

Thanks in advance :-)

### CODE REVIEW

**Comment by MattSturgeon** [comment]
Thanks for looking into this! We are already planning to tackle this one, see #2112.

In addition to just adding the plugin, we also need to add an option to `mini.nvim` to enable emulating the nvim-web-devicons API and do a treewide migration away from `plugins.*.iconsPackage` to instead document/warn users to enable either this plugin or `mini.nvim` with the API emulation.

Those extra steps could be handled in a separate PR though. I'm just mentioning them here as an FYI.

File: plugins/utils/web-devicons.nix [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{ lib, ... }:
+let
+  inherit (lib.nixvim) defaultNullOpts;
+in
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "web-devicons";
+  originalName = "nvim-web-devicons";
```
Comment (MattSturgeon) [inline plugins/utils/web-devicons.nix:7]: > `module 'web-devicons' not found`

```suggestion
  originalName = "nvim-web-devicons";
  luaName = "nvim-web-devicons";
```

File: tests/test-sources/plugins/utils/web-devicons.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+  empty = {
+    plugins.web-devicons.enable = true;
+  };
+
+  default = {
+    plugins.web-devicons = {
+      enable = true;
+    };
+  };
```
Comment (MattSturgeon) [inline tests/test-sources/plugins/utils/web-devicons.nix:10]: If this is identical to `empty`, there's no point including it really.

If you want to test some known defaults, that's fine though, even if we aren't declaring nix options for them.

**Comment by khaneliman** [comment]
> Thanks for looking into this! We are already planning to tackle this one, see #2112.
> 
> In addition to just adding the plugin, we also need to add an option to `mini.nvim` to enable emulating the nvim-web-devicons API and do a treewide migration away from `plugins.*.iconsPackage` to instead document/warn users to enable either this plugin or `mini.nvim` with the API emulation.
> 
> Those extra steps could be handled in a separate PR though. I'm just mentioning them here as an FYI.

Yeah, when I saw the PR I figured we could just append that stuff after the initialization. 

```bash
ERROR: Error detected while processing /nix/store/0vq4y5bpa2vmixafnd6gdjqcm43dbi6k-init.lua:
       > E5113: Error while calling lua chunk: /nix/store/0vq4y5bpa2vmixafnd6gdjqcm43dbi6k-init.lua:10: module 'web-devicons' not found:
       >     no field package.preload['web-devicons']
```
Looks like module isn't configured properly, yet. Need to either change name or set luaName, as suggested

**Review by MattSturgeon — COMMENTED** [review]
We still need a test adding, but otherwise looks good.

When done, are you able to squash/rebase? If not we can handle.

**Comment by refaelsh** [comment]
> We still need a test adding, but otherwise looks good.
> 
> When done, are you able to squash/rebase? If not we can handle.

I deleted the tests. You said, a few posts above, that:
```
If this is identical to empty, there's no point including it really.

If you want to test some known defaults, that's fine though, even if we aren't declaring nix options for them.
```
I concluded from this I should delete the tests. Did I misunderstood?

**Comment by MattSturgeon** [comment]
> You said, a few posts above, that:
> 
> 
> > If this is identical to `empty`, there's no point including it really.
> >
> > If you want to test some known defaults, that's fine though, even if we aren't declaring nix options for them.
> 
> 
> I concluded from this I should delete the tests. Did I misunderstood?

I meant that we should have one test, named "empty". The other two tests were identical to it, and therefore redundant.

Having _one_ empty test is not redundant, since it will check enabling the plugin works and no errors are printed while running neovim.

**Comment by refaelsh** [comment]
> > You said, a few posts above, that:
> > > If this is identical to `empty`, there's no point including it really.
> > > If you want to test some known defaults, that's fine though, even if we aren't declaring nix options for them.
> > 
> > 
> > I concluded from this I should delete the tests. Did I misunderstood?
> 
> I meant that we should have one test, named "empty". The other two tests were identical to it, and therefore redundant.
> 
> Having _one_ empty test is not redundant, since it will check enabling the plugin works and no errors are printed while running neovim.

Yes, I now see I misundertood. Sorry.
Resolved.

**Review by MattSturgeon — APPROVED** [review]
Everything looks good. Thanks for your persistence!

Are you able to squash or shall I ask mergify to do it?

**Comment by MattSturgeon** [comment]
@mergifyio squash first-commit

**Comment by MattSturgeon** [comment]
@mergifyio rebase

**Comment by MattSturgeon** [comment]
Hm it seems mergify attributes the commit to me, so you don't get any credit.

If you can run the following, that'd be great:

```nix
git fetch --all
git reset --hard origin/add-nvim-web-devicons
git reset --soft HEAD^
git commit --reuse-message HEAD@{1}
git push --force-with-lease
```

**Comment by refaelsh** [comment]
> If you can run the following, that'd be great:

I've run those commands, and now there is a problem, here is a screenshot:

Looks like the local branch and the remote branch have diverged.

**Comment by refaelsh** [comment]
> Thanks for your persistence!

No no, thank you for the reviewing! :-)

**Comment by khaneliman** [comment]
> > If you can run the following, that'd be great:
> 
> I've run those commands, and now there is a problem, here is a screenshot:  Looks like the local branch and the remote branch have diverged.

The point of the commands is to rewrite the git history and force push it up to overwrite whats on github.

**Comment by khaneliman** [comment]
Just did an amend commit to change author 
image

You can still force push changes yourself, as well.

**Comment by MattSturgeon** [comment]
@mergifyio queue

**Pull request merged**